src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -95,19 +96,9 @@
     private static final boolean ENABLE_SVC
         = cfg.getBoolean(ENABLE_SVC_PNAME, false);
 
-    /**
-     * Creates encodings.
-     *
-     * @param track the track that will own the temporal encodings.
-     * @param primary the array of the primary SSRCs for the simulcast streams.
-     * @param rtx the array of the RTX SSRCs for the simulcast streams.
-     * @param spatialLen the number of spatial encodings per simulcast stream.
-     * @param temporalLen the number of temporal encodings per simulcast stream.
-     * @return an array that holds the simulcast encodings.
-     */
     private static RTPEncodingDesc[] createRTPEncodings(
-        MediaStreamTrackDesc track, long[] primary, long[] rtx,
-        int spatialLen, int temporalLen)
+        MediaStreamTrackDesc track, Long[] primary,
+        int spatialLen, int temporalLen, Map<Long, List<Pair<Long, String>>> secondarySsrcs)
```

Comment:
This gets confusing quickly. How about an abstraction at some level, e.g. List<Pair<Long, String>> becomes SecondarySsrcs, which would leave us with a Map<Long, SecondarySsrcs>

Reply:
yeah i agree.  you're talking about making a SecondarySsrcs inner class, right?

Reply:
Now that I've read the rest of this I tend to like the following better: replace the abstract Pair class with a SecondarySsrc class in libjitsi. Here we'd have Map<Long, List<SecondarySsrc>> which isn't too bad, and we can pass SecondarySsrc as a single parameter to methods. But I don't have any strong preference either way.

Reply:
But I think it makes sense to define one structure in libjitsi and use it everywhere.

Reply:
i just implemented it with both `SecondarySsrc` and `SecondarySsrcs` classes and it isn't bad.  but after looking at it, i'm not sure i like passing `SecondarySsrc` into `RTPEncodingDesc` (as opposed to the individual arguments) because then we'd have to make `SecondarySsrc` accessible elsewhere and it doesn't seem worth it to have it be more than an internal helper and leaking it into a broader scope.  I'll push this change and you can let me know what you think.